### PR TITLE
raftstore: add request wait time and process time metrics

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR tries to fix issue #1562

### CODE REVIEW

**Comment by hhkbp2** [comment]
PTAL @siddontang @BusyJay @zhangjinpeng1987

File: src/raftstore/store/metrics.rs [hunk 1]
```diff
@@ -156,4 +156,18 @@ lazy_static! {
             "Histogram of keys written for regions",
              exponential_buckets(1.0, 2.0, 20).unwrap()
         ).unwrap();
+
+    pub static ref REQUEST_WAIT_TIME_HISTOGRAM: Histogram =
+        register_histogram!(
+            "tikv_request_wait_time_duration_secs",
```
Comment (siddontang) [inline src/raftstore/store/metrics.rs]: tikv_raftstore_request

File: src/raftstore/store/store.rs [hunk 2]
```diff
@@ -1869,7 +1871,9 @@ impl<T: Transport, C: PdClient> mio::Handler for Store<T, C> {
                     error!("{} handle raft message err: {:?}", self.tag, e);
                 }
             }
-            Msg::RaftCmd { request, callback } => self.propose_raft_command(request, callback),
+            Msg::RaftCmd { send_time, request, callback } => {
+                self.propose_raft_command(send_time, request, callback)
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: seem that you can record the wait time here directly and no need to pass to propose_raft_command

File: src/raftstore/store/store.rs [hunk 3]
```diff
@@ -113,6 +113,9 @@ pub struct Store<T, C: 'static> {
 
     region_written_bytes: LocalHistogram,
     region_written_keys: LocalHistogram,
+
+    request_wait_time: LocalHistogram,
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: we should move these to the RaftMetrics

File: src/raftstore/store/peer.rs [hunk 4]
```diff
@@ -671,7 +671,10 @@ impl Peer {
         }
     }
 
-    pub fn handle_raft_ready_apply(&mut self, mut ready: Ready, result: &mut ReadyResult) {
+    pub fn handle_raft_ready_apply(&mut self,
+                                   mut ready: Ready,
+                                   result: &mut ReadyResult,
+                                   request_process_time: &mut LocalHistogram) {
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: you may pass RaftMetric, so we can add other metric later.

File: src/raftstore/store/peer.rs [hunk 5]
```diff
@@ -1333,6 +1349,9 @@ impl Peer {
         cmd_resp::bind_term(&mut resp, self.term());
         cb.call_box((resp,));
 
+        let d = clocktime::raw_now() - lease_renew_time;
+        request_process_time.observe(duration_to_sec(d.to_std().unwrap()) as f64);
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: This will be conflicted with async-apply, maybe we should add this metric later.

Comment (siddontang) [inline src/raftstore/store/peer.rs]: /cc @BusyJay 

I think it is better to consider the request process time in async-apply, now only wait_time is ok.

**Comment by ngaut** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**